ConnectionFactory.php

```diff
@@ -45,8 +45,16 @@ public function createConnection(array $params, Configuration $config = null, Ev
             $this->initializeTypes();
         }
 
-        if (! isset($params['pdo']) && ! isset($params['charset'])) {
+        $connectionOverrideOptions = [];
+
+        if (isset($params['connection_override_options'])) {
+            $connectionOverrideOptions = $params['connection_override_options'];
+            unset($params['connection_override_options']);
```

Comment:
Could we do:

```php
$connectionOverrideOptions = $params['connection_override_options'] ?? [];
unset($params['connection_override_options']);
```

It's harmless to unset a key that exists.

Reply:
done